.github/workflows/scripts/build_helper.sh

```diff
@@ -107,7 +107,8 @@ fi
 NVCC_VER="$(nvcc --version | tail -n1 | sed -E 's/^Build //')"
 
 # Version from src/params.h
-MFAKTC_VER="$(grep -Eo '#define MFAKTC_VERSION "([0-9]\.[0-9]+\.[-0-9a-z]+)"' src/params.h | cut -d '"' -f 2)"
+# Match semver: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+MFAKTC_VER="$(LC_ALL=en_US.utf8 grep -Po '#define MFAKTC_VERSION "(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?"' src/params.h | cut -d '"' -f 2)"
```

Comment:
I think it's a good idea to check for one or more spaces as opposed to a single space. For example, something like `'#define MFAKTC_VERSION  "0.23.3"` would also be a valid C preprocessor directive.